packages/fastify-renderer/test/FastifyRenderer.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { FastifyRendererPlugin } from '../src/node/Plugin';
+import { ReactRenderer } from '../src/node/renderers/react/ReactRenderer';
+import { kRenderOptions } from "../src/node/symbols";
+import { newFastify } from "./helpers";
+
+describe('FastifyRenderer', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should throw if the fastify-renderer plugin was already registered in the same fastify context', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    await expect(server.register(FastifyRenderer)).rejects.toThrow("The decorator 'vite' has already been added!")
+  });
+
+  test('should throw if the fastify-renderer plugin was already registered in a different context', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    await expect(server.register(async instance => instance.register(FastifyRenderer))).rejects.toThrow()
+  });
+
+  test('should decorate the fastify instance with a "setRenderConfig" method', async () => {
+    const server = await newFastify()
+    await server.register(FastifyRenderer)
+    expect(server.setRenderConfig).toBeInstanceOf(Function)
+  });
+
+  test('should set default render config for the instance', async () => {
+    const server = await newFastify()
+
+    expect(server[kRenderOptions]).toBeUndefined()
+    await server.register(FastifyRenderer)
+    expect(server[kRenderOptions]).toBeInstanceOf(Object)
+  });
+
+  test('should set the renderOptions on the new fastify instance context', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    await server.register(async instance => {
+      expect(server[kRenderOptions]).toEqual(instance[kRenderOptions])
+    })
+  });
+
+  test('should mount vite routes at a prefix to avoid collision with user routes', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    expect(server.printRoutes()).toMatch('/.vite/')
+  });
+
+  // test.skip('should use config from vite dev server in dev mode', async () => {
+  // });
+
+  test('should close vite devServer when fastify server is closing in dev mode', async () => {
+    const devServer = await Vite.createServer();
+    const closeSpy = jest.spyOn(devServer, 'close')
+    jest.spyOn(Vite, 'createServer').mockImplementation(async () => devServer)
+
+    const server = await newFastify()
+    await server.register(FastifyRenderer, { devMode: true })
+    await server.listen(0)
+    await server.close()
+
+    expect(closeSpy).toHaveBeenCalled()
+  });
+
+  test('should prepare routes', async () => {
+    const server = await newFastify()
+    const prepareSpy = jest.spyOn(ReactRenderer.prototype, 'prepare')
+
+    await server.register(FastifyRenderer)
+    await server.listen(0)
+    await server.close()
+
+    expect(prepareSpy).toHaveBeenCalled()
+  });
```

Comment:
I think this is testing implementation a little too closely, and instead, I would try to test the interface that these things expose. If you feel like this is really important, you could add a thing where the renderer exposes which routes it knows about, and ask it. That test would be more robust to the way that the routes are added or prepared over time while still ensuring that something knows about them. And then again its a style thing, but I would say that it doesn't really matter if the renderer knows about the routes, what really matters is if those routes make it to the client! So, I'd probably say this is well covered by an end to end test.